lib/jerboa/format/body/attribute/lifetime.ex

```diff
@@ -0,0 +1,37 @@
+defmodule Jerboa.Format.Body.Attribute.Lifetime do
+  @moduledoc """
+  LIFETIME attribute as defined in [TURN RFC](https://trac.tools.ietf.org/html/rfc5766#section-14.2)
+  """
+
+  alias Jerboa.Format.Body.Attribute
+  alias Jerboa.Format.Lifetime.LengthError
+
+  defstruct duration: 0
+
+  @max_duration :math.pow(2, 32) - 1
+
+  @typedoc """
+  Represents a lifetime of the allocation
+
+  * `:duration` is a duration of a lifetime in seconds
+  """
+  @type t :: %__MODULE__{
+    duration: non_neg_integer
+  }
+
+  @doc false
+  @spec encode(t) :: binary
+  def encode(%__MODULE__{duration: duration})
+    when is_integer(duration) and duration <= @max_duration and duration >= 0 do
```

Comment:
Hm, I'm concerned about performance here. As I remember `in/2` macro rewrites expression above into:
```elixir
duration === 0 or duration === 1 or ... or duration === (a lot)
```
So we have 2^32 conditions executed lineary until we find a match. I'll take a look at implementation of `in/2`.